src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -224,8 +227,95 @@ package experimental {
         case _ => false
       }
     }
+
+    /**
+     * Calculate the inverse of this pattern set.
+     * 
+     * @return A BitSet matching all value (of the given with) iff it doesn't match this pattern.
+     */
+    def inverse: BitSet = {
+      val total = BitPat("b" + ("?".repeat(this.getWidth)))
+      total.subtract(this)
+    }
+
+    /**
+      * Calculate lowest posible value to match this pattern seset.
+      *
+      * @throws UnsupportedOperationException if this this is an empty pattern set.
+      * @return The matched value
+      */
+    def lowerbound: BigInt = terms.map(_.lowerbound).min
```

Comment:
I think we should private this, since I have no idea the use case.
If you can give some common use cases, we can public at it is.

Reply:
The reason this method is `public` is that another class (BitSetRangeIterator) need to invoke it. If we want to hide this method, we need to extract it into a `private object BitSetUtils`.

Reply:
Changed to private in 0c30bd61